radixdlt/src/main/java/com/radixdlt/network/messaging/MessageDispatcher.java

```diff
@@ -213,4 +222,31 @@ private SendResult updateStatistics(SendResult result) {
 			throw new UncheckedIOException("While serializing message", e);
 		}
 	}
+
+	private String hostId(Peer peer) {
+		return peer.supportedTransports()
+			.findFirst()
+			.map(ti -> String.format("%s:%s", ti.name(), ti.metadata()))
+			.orElse("None");
+	}
+
+	/**
+	 * Return true if we already have information about the given peer being banned.
+	 * Note that if the peer is already banned according to our address book, the
+	 * specified peer instance will have it's banned timestamp updated to match the
+	 * known peer's banned time.
+	 *
+	 * @param peerNid the corresponding node ID of the peer
+	 * @param messageType the message type for logging ignored messages
+	 * @return {@code true} if the peer is currently banned, {@code false} otherwise
+	 */
+	private boolean checkPeerBanned(EUID peerNid, String messageType) {
+		return this.addressBook.peer(peerNid)
+			.filter(kp -> kp.getTimestamp(Timestamps.BANNED) > this.timeSource.currentTime())
+			.map(kp -> {
```

Comment:
This seems like a counter-intuitive place to do the actual logging, I would expect that to be above the part that returns in the calling method.

Reply:
Yeah, I know.  I wanted to include the existing peer in the log message, and returning it elsewhere to allow logging to be more consistent seemed worse than the inconsistency to me.